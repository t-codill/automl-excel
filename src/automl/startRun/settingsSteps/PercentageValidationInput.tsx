import * as React from "react";
import { FormTextInput } from "../../components/Form/FormTextInput";
import { Validators } from "../../components/Form/Validators";
import { LabelWithCallout } from "../../components/Label/LabelWithCallout";
import { ISettingsStepParams } from "./ISettingsStepParams";

const percentageValidationInput: React.FunctionComponent = () => {
    return <div className="ms-Grid" dir="ltr">
        <div className="ms-Grid-col ms-sm12 ms-md6 ms-lg6 ms-xl6">
            <LabelWithCallout required={true} htmlFor={"percentageOfValidationData"} labelText="% validation of data">
                Percentage of validation data
            </LabelWithCallout>
        </div>
        <div className="ms-Grid-col ms-sm12 ms-md6 ms-lg6 ms-xl6">
            <FormTextInput<ISettingsStepParams, "percentageValidation">
                field="percentageValidation"
                placeholder="Percentage Validation data"
                ariaLabel="Please enter the number of percentage data"
                type="number"
                validators={[
                    Validators.required("Percentage of validation data is required"),
                    Validators.isInteger("Percentage of validation data should be an integer"),
                ]}
                defaultFormValue="20"
            />
        </div>
    </div>;
};

export { percentageValidationInput as PercentageValidationInput };
