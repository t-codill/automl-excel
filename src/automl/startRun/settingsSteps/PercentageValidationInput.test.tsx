import { shallow } from "enzyme";
import * as React from "react";
import { PercentageValidationInput } from "./PercentageValidationInput";

describe("Percentage Validation Input", () => {
    it("should render", () => {
        const tree = shallow(
            <PercentageValidationInput />
        );
        expect(tree)
            .toMatchSnapshot();
    });
});
