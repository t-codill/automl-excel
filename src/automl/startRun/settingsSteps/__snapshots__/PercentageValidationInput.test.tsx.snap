// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Percentage Validation Input should render 1`] = `
<div
  className="ms-Grid"
  dir="ltr"
>
  <div
    className="ms-Grid-col ms-sm12 ms-md6 ms-lg6 ms-xl6"
  >
    <LabelWithCallout
      htmlFor="percentageOfValidationData"
      labelText="% validation of data"
      required={true}
    >
      Percentage of validation data
    </LabelWithCallout>
  </div>
  <div
    className="ms-Grid-col ms-sm12 ms-md6 ms-lg6 ms-xl6"
  >
    <FormTextInput
      ariaLabel="Please enter the number of percentage data"
      defaultFormValue="20"
      field="percentageValidation"
      placeholder="Percentage Validation data"
      type="number"
      validators={
        Array [
          [Function],
          [Function],
        ]
      }
    />
  </div>
</div>
`;
